pkg/operator/options/options_validation.go

```diff
@@ -29,22 +29,45 @@ func (o Options) Validate() error {
 	validate := validator.New()
 	return multierr.Combine(
 		o.validateRequiredFields(),
-		o.validateEndpoint(),
+		o.validateClusterEndpoint(),
+		o.validateArmAuthMethod(),
 		o.validateVMMemoryOverheadPercent(),
 		o.validateVnetSubnetID(),
 		validate.Struct(o),
 	)
 }
 
-func (o Options) validateVnetSubnetID() error {
-	_, err := utils.GetVnetSubnetIDComponents(o.SubnetID)
-	if err != nil {
-		return fmt.Errorf("vnet-subnet-id is invalid: %w", err)
+func (o Options) validateRequiredFields() error {
+	if o.Location == "" {
+		return fmt.Errorf("missing field, location")
+	}
+	if o.SubscriptionID == "" {
+		return fmt.Errorf("missing field, subscription-id")
+	}
+	if o.ClusterEndpoint == "" {
+		return fmt.Errorf("missing field, cluster-endpoint")
+	}
+	if o.ClusterName == "" {
+		return fmt.Errorf("missing field, cluster-name")
 	}
+
+	if o.NodeResourceGroup == "" {
+		return fmt.Errorf("missing field, node-resource-group")
```

Comment:
same comment about shared error type. we really only care that 

1. its missing 
2. what is missing